### DISCUSSION TITLE
Managing NATS Instance Scaling: Automatically Updating Configuration for New Instances

### DISCUSSION THREAD

**Discussion by mohamedsaleem18** [nats-io/nats-server#5892] [Q&A]
**Consumer connection url**
A NATS cluster currently consists of 3 NATS instances. According to recommendations from NATS.io, consumers should connect to the cluster using the seed URLs of each NATS instance to ensure automatic failover.

If the cluster is scaled up to 5 instances, is there a method for consumers to automatically detect the new NATS instances and update the NATS URLs accordingly?

**For example:**
       `nats://localhost:31420,nats://localhost:31421,nats://localhost:31422`

**Cluster configuration**
In the nats-server.config file, the URLs (routes) of each NATS instance are specified explicitly. When scaling the NATS instances in the StatefulSet from 3 to 5, is there a way to automatically update the nats-server.config file with the new instance URLs (routes)?

**For example:**
  ```
routes = [
      nats://nats-0.nats-21019-headless:6222
      nats://nats-1.nats-21019-headless:6222
      nats://nats-2.nats-21019-headless:6222
  ]
```

**Comment by wallyqs** [comment]
The helm chart should do this for you, if you use the `nats-io/k8s` that would do it when setting the number of replicas.  We will report to the bitnami team that the output from the bitnami/nats chart is wrong here: https://github.com/bitnami/charts/blob/main/bitnami/nats/values.yaml#L251-L260

**Reply by mohamedsaleem18** [reply]
Thank you.  How about the consumer connecting to the NATS cluster using seed url in the connection url string?

nats://localhost:31420,nats://localhost:31421,nats://localhost:31422

**Reply by jnmoyne** [reply]
Once a client connects to one of the servers listed in it's connection URL it will then automatically learn about other servers that may be present in the cluster (this is using an internal 'gossip' protocol, which can be optionally turned off if that behavior is unwanted).

**Reply by mohamedsaleem18** [reply]
Thank you for the clarification.
